python/plugins/processing/algs/gdal/ogr2ogr.py

```diff
@@ -99,6 +99,10 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         input_details = self.getOgrCompatibleSource(
             self.INPUT, parameters, context, feedback, executing
         )
+        if not input_details.layer_name:
+            raise QgsProcessingException(
+                self.invalidSourceError(parameters, self.INPUT)
+            )
```

Comment:
This should be after we've set convertAllLayers, and the condition should be:

    if not convertAllLayers and not input_details.layer_name: